Move PersistenceBeanTest into fat test folder

Azquelt opened the pull request:
Fixes #56

idlewis commented:
I think the commit needs a better comment, or it should be split up. There are multiple changes here, most completely unrelated to moving PersitenceBeanTest

Azquelt commented:
They're not totally unrelated but it wasn't as straightforward as I'd hoped. PersistenceBean used TestUtils which was also used by other unit tests so I couldn't just move it. In the end I moved one of the methods into the test-utils project and removed the other entirely as it duplicated logic we already had elsewhere.

I'll have a go at splitting it up so that it's easier to follow.

Azquelt committed:
Add method for matching multiple methods by id

This allows us to check that a collection contains the given list of
assets (as identified by their ids) without getting an unsafe type
warning.

can now use `contains(assetsWithIds(asset1, asset2))`

rather than `contains(hasId(asset1), hasId(asset2))`

The latter will cause a warning because contains() uses varargs and
would create an array of Matcher<? super T>, but arrays don't support
generics.

Azquelt committed:
Move PersistenceBeanTest into fat test folder

This required moving the methods in TestUtils, as they were used both by
the PersistenceBeanTest and other unit tests.

slurp() was moved to BasicChecks in test-utils, which makes it available
to all tests.

assertAssetList was removed and replaced with calls to assertThat()
using the ServerAssetByIdMatcher

Azquelt commented:
Commits split, hopefully the commit messages are more clear

Build passed successfully
